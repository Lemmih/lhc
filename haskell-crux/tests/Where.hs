module Basic1 where

x = y
  where y = x

fn1 0 = y
  where
    y = 10

fn2 n = y where y = fn2 n

fn3 n = even
  where
    even = 0 : odd
    odd = 1 : even
