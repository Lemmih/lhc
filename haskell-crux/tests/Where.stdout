Basic1.x : ∀ v18. v18
Basic1.x =
  λ @.v18^1:* → 
  let
    Basic1.y^2:v18 = Basic1.x
                       @.v18^1
  in
    Basic1.y^2
Basic1.fn1 : LHC.Prim.Int →  LHC.Prim.Int
Basic1.fn1 =
  λ arg^3:LHC.Prim.Int → 
  let
    Basic1.y^4:LHC.Prim.Int = LHC.Prim.I#
                                (LHC.Prim.i64toi32 10)
  in
    scrut^5 ← arg^3
    LHC.Prim.I# i^6:LHC.Prim.I32 ← scrut^5:LHC.Prim.Int
    i64^7 ← LHC.Prim.i32toi64 i^6
    0 ← i64^7:LHC.Prim.I64
    Basic1.y^4
Basic1.fn2 : ∀ v11 v10. v11 →  v10
Basic1.fn2 =
  λ @.v11^8:* @.v10^9:* Basic1.n^10:v11 → 
  let
    Basic1.y^11:v10 = Basic1.fn2
                            @.v11^8 @.v10^9 Basic1.n^10
  in
    Basic1.y^11
Basic1.fn3 : ∀ v1. v1 →  [LHC.Prim.Int]
Basic1.fn3 =
  λ @.v1^12:* Basic1.n^13:v1 → 
  let
    Basic1.even^14:[LHC.Prim.Int] = LHC.Prim.Cons
                                        (LHC.Prim.I# (LHC.Prim.i64toi32
                                            0)) Basic1.odd^15
    Basic1.odd^15:[LHC.Prim.Int] = LHC.Prim.Cons
                                       (LHC.Prim.I# (LHC.Prim.i64toi32
                                           1)) Basic1.even^14
  in
    Basic1.even^14
